api-clients/HttpClientConsoleApp/HttpClientConsoleApp/Entities/Product.cs

```diff
@@ -0,0 +1,28 @@
+﻿namespace HttpClientConsoleApp.Entities;
+
+class Product
+{
+    public int ProductId { get; set; }
+
+    public string ProductName { get; set; }
+
+    public int? SupplierId { get; set; }
+
+    public int? CategoryId { get; set; }
+
+    public string QuantityPerUnit { get; set; }
+
+    public decimal? UnitPrice { get; set; }
+
+    public short? UnitsInStock { get; set; }
+
+    public short? UnitsOnOrder { get; set; }
+
+    public short? ReorderLevel { get; set; }
+
+    public bool Discontinued { get; set; }
+
+    public virtual Category? Category { get; set; }
```

Comment:
Ain't sure you need to add `virtual` modifier here since your HTTP client is a different app and doesn't know anything about ORM used by server and also you are not using any overrides for it.
And also, since `Category` is a reference type, do you need this nullable cast?